src_code/src/controllers/services/Inventory/GetAllModels.js

```diff
@@ -0,0 +1,65 @@
+/** Initialize neccessary modules */
+const responseBuilder = require("../../../utils/interfaces/IResponseBuilder");
+const db = require("../../../configurations/database/DatabaseConfigurations");
+
+/**
+ * Retrieves information about all equipment models and their respective equipment counts.
+ *
+ * @param {Object} res - Express response object for sending HTTP responses.
+ * @returns {Object} An HTTP response containing information about equipment models and their counts.
+ * If successful, returns a JSON response with status 200 and an array of model information.
+ * If no models are found, returns a JSON response with status 400 and an error message.
+ * If an error occurs during the process, returns a JSON response with status 503 and an error message.
+ */
+async function GetAllModels(res, req) {
+    try {
+        /** Retrieve all models information */
+        const allModels = await db("equipment_model")
+            .select(
+                "equipment_type.TYPE_NAME as typeName",
+                "equipment_model.FK_TYPE_ID AS typeId",
+                "equipment_model.PK_MODEL_ID as modelId",
+                "equipment_model.MODEL_NAME as modelName",
+                "equipment_model.MODEL_PHOTO_URL as photoUrl"
+            )
+            .join("equipment_type", "equipment_type.PK_TYPE_ID", "=", "equipment_model.FK_TYPE_ID");
+
+        /** Check if there are no models, return a BadRequest response */
+        if (!allModels || allModels?.length === 0) {
+            return responseBuilder.BadRequest(res, "There is no model added yet.");
```

Comment:
Is there a reason this is a BadRequest error? I don't believe this is due to an error in the request.

Reply:
See comments above. if models length is 0, then no object/array will be returned, it makes easier process for front end to implement and handle message.